speculation-rules.bs

```diff
@@ -627,6 +629,97 @@ A <dfn>prerender candidate</dfn> is a [=struct=] with the following [=struct/ite
   1. [=Assert=]: This step is not reached.
 </div>
 
+<h2 id="content-security-policy">Content-Security-Policy</h2>
+<h3 id="content-security-policy-patches">Required Spec Patches</h3>
+Inline script use is also restricted by the [[CSP]], but there is a case a developer doesn't want to permit general inline scripts, but want to permit inline speculation rules. So, we make the following change that extends [[CSP]] to support a new source keyword for inline speculation rules.
+
+<h3 id="content-security-policy-patches-directives">Directives</h3>
+In the [[CSP#framework-directives]], we define several associated algorithms, and an <dfn dfn-type="grammar" link-type="grammar">inline check</dfn> algorithm needs a patch to define how to choose the type argument.
```

Comment:
Here and throughout, you can use `<dfn grammar>` instead of `<dfn dfn-type="grammar" link-type="grammar">`.

Also, you should use `noexport` on all definitions that are duplicates of definitions from the CSP spec. So in total, `<dfn grammar noexport>`, or `<dfn for="directive" noexport>`.